odl/space/weighting.py

```diff
@@ -686,25 +679,30 @@ def __init__(self, const, impl, exponent=2.0, dist_using_inner=False):
         """
         super().__init__(impl=impl, exponent=exponent,
                          dist_using_inner=dist_using_inner)
-        self._const = float(const)
+        self.__const = float(const)
         if self.const <= 0:
             raise ValueError('expected positive constant, got {}'
                              ''.format(const))
         if not np.isfinite(self.const):
             raise ValueError('`const` {} is invalid'.format(const))
 
+    def __hash__(self):
```

Comment:
Prefer if `__hash__` is always right after `__eq__` since they are so strongly connected.